CMakeLists.txt

```diff
@@ -170,14 +170,27 @@ if (NOT HAERO_ENABLE_GPU AND NOT APPLE)
   endif()
 endif()
 
+set(HAERO_AMD_GPU 0)
+# NOTE: this is hacky, but done as a result of not knowing what the
+# Kokkos_ARCH_$DEVICE_ARCH variable will be named
+string(FIND "${HAERO_DEVICE_ARCH}" "AMD" AMD_GPU)
```

Comment:
I think you could replace lines 176-177 with `if (HAERO_DEVICE_ARCH MATCHES "AMD")`.